guides/session.rst

```diff
@@ -1,76 +1,113 @@
+    to see which features are supported by each driver.
 
 Basic Browser Interaction
-~~~~~~~~~~~~~~~~~~~~~~~~~
+-------------------------
 
-After you've instantiated the ``$session`` object, you can control the actual
-browser emulator with it:
+Now that your session is started, you'll want to open a page with it. Just
+after starting, the session is not on any page (in a real browser, you would
+on the ``about:blank`` page), and calling any other action is likely to fail.
 
 .. code-block:: php
 
-    // open some page in browser:
     $session->visit('http://my_project.dev/some_page.php');
 
-    // get the current page URL:
-    echo $session->getCurrentUrl();
+.. note::
 
-    // get the response status code:
-    echo $session->getStatusCode();
+    Mink is primarily designed to be used for testing websites. To allow
+    you to browse and test error pages, the ``Session::visit`` method does
+    not consider error status codes as invalid. It will *not* throw an exception
+    in this case. You will need to check the status code (or certain text
+    on the page) to know if the response was successful or not.
+
+Interacting with the Page
+-------------------------
+
+The session gives you access to the page through the ``Session::getPage``
+method. This allows you to :doc:`traverse </guides/traversing-pages>` and
+:doc:`manipulate it </guides/manipulating-pages>`. The next chapters cover
+the page API in depth. Most of what you'll do with Mink will use this object,
+but you can continue reading to learn more about the Session.
 
-    // get page content:
-    echo $session->getPage()->getContent();
+Using the Browser History
+-------------------------
 
-    // open another page:
-    $session->visit('http://my_project.dev/second_page.php');
+The session gives you access to the browser history:
+
+.. code-block:: php
+
+    // get the current page URL:
+    echo $session->getCurrentUrl();
 
     // use history controls:
     $session->reload();
     $session->back();
     $session->forward();
 
-    // evaluate JS expression:
-    echo $session->evaluateScript(
-        "return 'something from browser';"
-    );
+Cookie Management
+-----------------
 
-    // wait for n milliseconds or
-    // till JS expression becomes true:
-    $session->wait(
-        5000,
-        "$('.suggestions-results').children().length > 0"
-    );
+The session can manipulate cookies available in the browser.
+
+.. code-block:: php
+
+    // set cookie:
+    $session->setCookie('cookie name', 'value');
+
+    // get cookie:
+    echo $session->getCookie('cookie name');
+
+    // delete cookie:
+    $session->setCookie('cookie name', null);
 
 .. note::
 
-    Although Mink does its best on removing browser differences between different
-    browser emulators, it can't do much in some cases. See the :ref:`driver-feature-support`
-    to see which features are supported by each driver.
+    With drivers that use a browser - like Selenium2 - you may be restricted
```

Comment:
That is both false and true at same time because fact, that driver is using real browser doesn't consider driver ways of controlling it. For example Selenium2 is controlling window manager of OS and literally can move mouse as user does and see all the cookies (his cookie management method allows to set httpOnly and other cookies: https://code.google.com/p/selenium/wiki/JsonWireProtocol#Cookie_JSON_Object). On the other side Sahi is controlling browser via JavaScript, which can't access HttpOnly cookies.

Maybe we should write it like this:
```
With drivers that use JavaScript to control the browser - like Sahi - you may 
be restricted to accessing/setting only `HttpOnly cookies`_ .
```

Maybe we should also note HttpOnly cookie restriction on the driver feature comparison page.